blt/src/Blt/Plugin/Commands/Tests/CypressTestCommand.php

```diff
@@ -0,0 +1,178 @@
+
+    $this->baseDir = $this->getConfigValue("tests.cypress.basePath");
+  }
+
+  /**
+   * Launch point for cypress tests.
+   *
+   * @command tests:cypress
+   * @description Execute all cypress tests.
+   *
+   * @usage -D tests.cypress.paths=${PWD}/tests/cypress/features/
+   *
+   * @aliases cypress tests:cypress:run tcr
+   */
+  public function cypressRunTestsMain() {
+    $commands = [
+      'tests:cypress:npm-install',
+      'tests:cypress:pretest',
+      'tests:cypress:runimpl'
+    ];
+
+    $this->invokeCommands($commands);
+  }
+
+  /**
+   * Set up to run cypress tests
+   *
+   * @command tests:cypress:npm-install
+   * @description Install the NPM packages needed in order to run cypress tests.
+   *
+   */
+  public function cypressSetup() {
+    // If we get tired of the amount of time it takes to do a complete clean install
+    // on every run for local tests, replace this with a call to taskNpmInstall.
+    $result = $this->taskNpmCleanInstall()->dir($this->baseDir)->run();
+    $exit_code = $result->getExitCode();
+
+    if($exit_code) {
+      throw new BltException('Cypress npm install failed.');
+    }
+  }
+
+  /**
+   * Run tests with side-effects the other tests depend on.
+   *
+   * @command tests:cypress:pretest
+   *
+   * @interactGenerateSettingsFiles
+   * @interactInstallDrupal
+   * @validateDrupalIsInstalled
+   * @launchWebServer
+   *
+   * @return void
+   */
+  public function cypressSetuptests() {
+    $pretest = $this->getConfigValue('tests.cypress.pretestScript');
+
+    if(!is_string($pretest)) {
+      throw new BltException('tests.cypress.pretestScript must be a string.');
+    }
+
+    $result = $this->taskNpmRunScript()
+        ->scriptName($pretest)
+        ->dir($this->baseDir)->run();
+    $exit_code = $result->getExitCode();
+
+    if($exit_code) {
+      throw new BltException('Cypress pre-test failed.');
+    }
+  }
+
+  /**
+   * Run cypress test directly, bypassing the setup steps.
+   *
+   * @command tests:cypress:runimpl
+   *
+   * @interactGenerateSettingsFiles
+   * @interactInstallDrupal
+   * @validateDrupalIsInstalled
+   * @launchWebServer
+   *
+   * @throws \Acquia\Blt\Robo\Exceptions\BltException
+   * @throws \Exception
+   */
+  public function cypressRunTests() {
+
+    $testScript = $this->getConfigValue("tests.cypress.testScript");
+
+    if(!is_string($testScript)) {
+      throw new BltException('tests.cypress.pretestScript must be a string.');
+    }
+
+    // Running cypress in parallel requires Xvfb.
+    $this->startXvfb();
+
+    $result = $this->taskNpmRunScript()
+        ->scriptName($testScript)
+        ->dir($this->baseDir)->run();
+
+    $this->killXvfb();
+
+    $exit_code = $result->getExitCode();
+
+    if($exit_code) {
+      throw new BltException('Cypress tests failed.');
+    }
+
+  }
+
+  /**
+   * Helper method to handle the details of launching Xvfb.
+   */
+  private function startXvfb() {
+
+    $port = CypressTestCommand::XVFB_PORT;
```

Comment:
You know, you could just make these attributes like @launchWebServer that can wrap the running of the command.

Reply:
That does simplify the logic a bit.  And gave me an idea for something similar with running `npm ci`.